t/020-basic.t

```diff
@@ -37,7 +37,7 @@ if library-exists('portaudio', v2) {
         ok $stream.active, "and it is active now";
         is $stream.info.sample-rate, 44100e0, "got some info";
         nok $stream.stopped, "and it is obviously not stopped anymore";
-        ok $stream.write-available, "and write-available gives us something";
+        cmp-ok $stream.write-available, '<=', 0, "and write-available gives us something non-negative";
```

Comment:
Typically if you are planning on making a PR it helps to make each one in a separate branch so you can keep any other changes you may make in the meantime separate.

I'll try to take a better look at the callback stuff later in the week.

Thanks again.